tests/unit/testsuite/Bolt/Boltpay/Model/Admin/ExtraConfigTest.php

```diff
@@ -0,0 +1,373 @@
+     *
+     * @covers ::save
+     * @covers ::hasValidPriceFaultTolerance
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     * @throws Exception when there is an error saving the extra config to the database
+     */
+    public function save_withInvalidOptions_savesOldValue()
+    {
+        $invalidPriceFaultTolerance = -1;
+        $value = json_encode(
+            array(
+                'priceFaultTolerance' => $invalidPriceFaultTolerance,
+                'boltPrimaryColor'    => '##',
+                'datadogKeySeverity'  => '##'
+            )
+        );
+        $oldValue = '{}';
+        $sessionMock = $this->getMockBuilder('Mage_Core_Model_Session_Abstract')
+            ->setMethods(array('addError'))
+            ->getMock();
+
+        Bolt_Boltpay_TestHelper::stubSingleton('core/session', $sessionMock);
+
+        $this->currentMock->expects($this->once())->method('getValue')->willReturn($value);
+        $this->currentMock->expects($this->once())->method('getOldValue')->willReturn($oldValue);
+        $this->currentMock->expects($this->once())->method('setValue')->with($oldValue);
+
+
+        $sessionMock->expects($this->once())->method('addError')
+            ->with(
+                "Invalid value for extra option `priceFaultTolerance`.[" . $invalidPriceFaultTolerance . "]
+                         A valid value must be a positive integer."
+            );
+
+        $this->currentMock->save();
+    }
+
+    /**
+     * @test
+     * that hasValidBoltPrimaryColor returns false for invalid color hex and adds error message to session
+     *
+     * @covers ::hasValidBoltPrimaryColor
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidBoltPrimaryColor_withInvalidColorHex_returnsFalseAndAddsErrorMessageToSession()
+    {
+        $invalidBoltPrimaryColorHex = "##";
+        $sessionMock = $this->getMockBuilder('Mage_Core_Model_Session_Abstract')
+            ->setMethods(array('addError'))
+            ->getMock();
+        Bolt_Boltpay_TestHelper::stubSingleton('core/session', $sessionMock);
+
+        $sessionMock->expects($this->once())->method('addError')
+            ->with(
+                "Invalid hex color value for extra option `boltPrimaryColor`. [" . $invalidBoltPrimaryColorHex . "] It must be in 6 or 8 character hex format.  (e.g. #f00000 or #3af508a2)"
+            );
+
+        $this->assertFalse($this->currentMock->hasValidBoltPrimaryColor($invalidBoltPrimaryColorHex));
+    }
+
+    /**
+     * @test
+     * that hasValidDatadogKeySeverity returns false for invalid severity keys and adds error message to session
+     *
+     * @covers ::hasValidDatadogKeySeverity
+     *
+     * @throws Mage_Core_Exception if unable to stub singleton
+     */
+    public function hasValidDatadogKeySeverity_withInvalidSeverityKey_returnsFalseAndAddsErrorMessageToSession()
```

Comment:
@mccottry Can you add a new unit test named hasValidDatadogKeySeverity_withValidSeverityKey_returnsTrue?

Reply:
I can update the test for you.